src/compiler/checker.ts

```diff
@@ -1013,6 +1020,20 @@ module ts {
             return type;
         }
 
+        function cloneInterfaceType(type: InterfaceType, flags: TypeFlags): InterfaceType {

```

Comment:
why do you need this. this changes every time we change the structure of Interface/ObjectTypes